### PR TITLE
Day5 Ex1 RecyclerView

### PR DESCRIPTION
**Pull request by at-thienngnuyen**
Ex 05 Thien Nguyen

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -23,8 +23,16 @@ dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
     androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
         exclude group: 'com.android.support', module: 'support-annotations'
+
```
Comment (at-thangtruong) [inline app/build.gradle]: Please remove this line

File: app/src/main/java/vn/asiantech/internship/Buoi5/Model/FriendsMyAdapter.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
+package vn.asiantech.internship.Buoi5.Model;
```
Comment (at-thangtruong) [inline app/src/main/java/vn/asiantech/internship/Buoi5/Model/FriendsMyAdapter.java]: Change the package to this `vn.asiantech.internship.friend;`

File: app/src/main/java/vn/asiantech/internship/Buoi5/Model/FriendsMyAdapter.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+import java.util.ArrayList;
+
+import vn.asiantech.internship.R;
+
+/**
+ * Created by root on 6/9/17.
+ */
+
+public class FriendsMyAdapter extends RecyclerView.Adapter<FriendsMyAdapter.MyViewHolder> {
```
Comment (at-thangtruong) [inline app/src/main/java/vn/asiantech/internship/Buoi5/Model/FriendsMyAdapter.java]: Change the class name like this `ListFriendAdapter`

File: app/src/main/java/vn/asiantech/internship/Buoi5/Model/FriendsMyAdapter.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+    private ArrayList<User> mUserArrayList;
```
Comment (at-thangtruong) [inline app/src/main/java/vn/asiantech/internship/Buoi5/Model/FriendsMyAdapter.java]: Please change this line to
```java
private List<Friend> mFriends;
```

File: app/src/main/java/vn/asiantech/internship/Buoi5/Model/FriendsMyAdapter.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    private Context mContext;
+
+    public FriendsMyAdapter(ArrayList<User> userArrayList, Context context) {
+        this.mUserArrayList = userArrayList;
+        this.mContext = context;
+    }
+
+    @Override
+    public MyViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
+        View v = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.friends_list_item_buoi5, viewGroup, false);
```
Comment (at-thangtruong) [inline app/src/main/java/vn/asiantech/internship/Buoi5/Model/FriendsMyAdapter.java]: Change the layout name to `item_list_friend`

File: app/src/main/java/vn/asiantech/internship/Buoi5/Model/User.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
+package vn.asiantech.internship.Buoi5.Model;
+
+/**
+ * Created by root on 6/9/17.
+ */
+
+public class User {
```
Comment (at-thangtruong) [inline app/src/main/java/vn/asiantech/internship/Buoi5/Model/User.java]: Change class name to `Friend`

File: app/src/main/java/vn/asiantech/internship/Buoi5/Model/User.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
+package vn.asiantech.internship.Buoi5.Model;
```
Comment (at-thangtruong) [inline app/src/main/java/vn/asiantech/internship/Buoi5/Model/User.java]: Change package to `package vn.asiantech.internship.models;`

File: app/src/main/java/vn/asiantech/internship/Buoi5/View/FriendFragment.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.ArrayList;
+
+import vn.asiantech.internship.Buoi5.Model.FriendsMyAdapter;
+import vn.asiantech.internship.Buoi5.Model.User;
+import vn.asiantech.internship.R;
+
+public class FriendFragment extends Fragment {
+
+
+    private ArrayList<User> mUserArrayList = new ArrayList<>();
```
Comment (at-thangtruong) [inline app/src/main/java/vn/asiantech/internship/Buoi5/View/FriendFragment.java]: Change to this `private List<Friend> mFriends = new ArrayList<>();`

File: app/src/main/java/vn/asiantech/internship/Buoi5/View/FriendFragment.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        View v = inflater.inflate(R.layout.friends_fragment, container, false);
```
Comment (at-thangtruong) [inline app/src/main/java/vn/asiantech/internship/Buoi5/View/FriendFragment.java]: Change layout name like this `fragment_list_friend`

File: app/src/main/java/vn/asiantech/internship/Buoi5/View/FriendFragment.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+import android.view.ViewGroup;
+
+
+import java.util.ArrayList;
+
+import vn.asiantech.internship.Buoi5.Model.FriendsMyAdapter;
+import vn.asiantech.internship.Buoi5.Model.User;
+import vn.asiantech.internship.R;
+
+public class FriendFragment extends Fragment {
```
Comment (at-thangtruong) [inline app/src/main/java/vn/asiantech/internship/Buoi5/View/FriendFragment.java]: Change class name like this `ListFriendFragment`

File: app/src/main/java/vn/asiantech/internship/models/Friend.java [hunk 11]
```diff
@@ -0,0 +1,41 @@
+package vn.asiantech.internship.models;
+
+/**
+ * Created by root on 6/9/17.
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/models/Friend.java]: remove this break line

File: app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+import vn.asiantech.internship.R;
+
+/**
+ * Created by root on 6/9/17.
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java]: remove this break line

File: app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+public class ListFriendAdapter extends RecyclerView.Adapter<ListFriendAdapter.MyViewHolder> {
+
+    private List<Friend> mFriends;
+    private Context mContext;
+
+    public ListFriendAdapter(List<Friend> FriendArrayList, Context context) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java]: change `FriendArrayList` to `friends`

File: app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+        //   MyViewHolder mViewHolder =  myViewHolder;
+        myViewHolder.mTvFriendName.setText(mFriends.get(i).getNameFriend());
+        myViewHolder.mIsFriend = mFriends.get(i).isFriend();
+        if (mFriends.get(i).isFriend()) {
+            myViewHolder.mBtnAdd.setBackground(mContext.getResources().getDrawable(R.drawable.friends_bg_btn_add_selected));
+            myViewHolder.mBtnAdd.setText(mContext.getResources().getString(R.string.Button_Text_Friend));
+            myViewHolder.mBtnAdd.setTextColor(mContext.getResources().getColor(R.color.friendsColorWhite));
+        }
+    }
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java]: remove a break line

File: app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+
+    @Override
+    public int getItemCount() {
+        return mFriends.size();
+    }
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java]: remove a break line

File: app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+
+    public class MyViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java]: add java doc

File: app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+        @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
+        private void unSetColor() {
+            mBtnAdd.setBackground(mContext.getResources().getDrawable(R.drawable.friends_bg_btn_add));
+            mBtnAdd.setText(mContext.getResources().getString(R.string.Button_Text_Add));
+            mBtnAdd.setTextColor(mContext.getResources().getColor(R.color.friendsColorBorder));
+            mFriends.get(getLayoutPosition()).setFriend(false);
+            mIsFriend = false;
+        }
+    }
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java]: remove a break line

File: app/src/main/java/vn/asiantech/internship/views/ListFriendFragment.java [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/views/ListFriendFragment.java]: remove a break line

File: app/src/main/java/vn/asiantech/internship/views/ListFriendFragment.java [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import vn.asiantech.internship.models.Friend;
+import vn.asiantech.internship.models.ListFriendAdapter;
+import vn.asiantech.internship.R;
+
+public class ListFriendFragment extends Fragment {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/views/ListFriendFragment.java]: add java doc

File: app/src/main/java/vn/asiantech/internship/views/ListFriendFragment.java [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/views/ListFriendFragment.java]: remove a break line

File: app/src/main/res/drawable/btn_background.xml [hunk 21]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (at-vinhhuynh) [inline app/src/main/res/drawable/btn_background.xml]: change file name to `bg_btn_add`

File: app/src/main/res/drawable/friends_bg_btn_add.xml [hunk 22]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (at-vinhhuynh) [inline app/src/main/res/drawable/friends_bg_btn_add.xml]: change file name to `bg_btn_add_friend`

File: app/src/main/res/drawable/friends_bg_btn_add_selected.xml [hunk 23]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (at-vinhhuynh) [inline app/src/main/res/drawable/friends_bg_btn_add_selected.xml]: change file name to `bg_btn_added_friend`

File: app/src/main/res/drawable/ic_menu.xml [hunk 24]
```diff
@@ -0,0 +1,7 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (at-vinhhuynh) [inline app/src/main/res/drawable/ic_menu.xml]: change file name `vector_menu`

File: app/src/main/res/layout/item_list_friend.xml [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+            android:stateListAnimator="@null"
+            android:text="@string/Button.Text.Add" />
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginLeft="@dimen/marginSomeThings"
+            android:layout_toRightOf="@id/imgAvatar"
+            android:layout_toLeftOf="@id/btnAdd"
+            android:id="@+id/linearLayout"
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/item_list_friend.xml]: change id to `lnParent`

File: app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java [hunk 26]
```diff
@@ -0,0 +1,106 @@
...
+    public ListFriendAdapter(List<Friend> friends, Context context) {
+        this.mFriends = friends;
+        this.mContext = context;
+    }
+
+    @Override
+    public MyViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
+        View v = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_list_friend, viewGroup, false);
+        MyViewHolder holder = new MyViewHolder(v);
+        holder.setIsRecyclable(false);
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java]: what is this method mean? `setIsRecyclable()`

File: app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java [hunk 27]
```diff
@@ -0,0 +1,106 @@
...
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+import java.util.List;
+import vn.asiantech.internship.R;
+
+/**
+ * Created by root on 6/9/17.
+ */
+public class ListFriendAdapter extends RecyclerView.Adapter<ListFriendAdapter.MyViewHolder> {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java]: Why create class adapter on `model` package?

File: app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java [hunk 28]
```diff
@@ -0,0 +1,106 @@
...
+    public int getItemCount() {
+        return mFriends.size();
+    }
+
+    /**
+     * Viewhoder class
+     */
+    public class MyViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        private TextView mTvFriendName;
+        private TextView mTvFriendDes;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java]: remove this variable if you don't use it

File: app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java [hunk 29]
```diff
@@ -0,0 +1,106 @@
...
+        private ImageView mImageView;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/models/ListFriendAdapter.java]: remove this variable if you don't use it

**Comment by at-vinhhuynh** [comment]
@at-thienngnuyen 
Please move image to `drawable-xxhdpi` and change name follow convention

File: app/src/main/java/vn/asiantech/internship/ListFriendAdapter.java [hunk 30]
```diff
@@ -0,0 +1,105 @@
...
+        this.mContext = context;
+    }
+
+    @Override
+    public MyViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
+        View v = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_list_friend, viewGroup, false);
+        return new MyViewHolder(v);
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/ListFriendAdapter.java]: remove this line `@RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN`

File: app/src/main/java/vn/asiantech/internship/ListFriendAdapter.java [hunk 31]
```diff
@@ -0,0 +1,105 @@
...
+        private boolean mIsFriend;
+
+        public MyViewHolder(View itemView) {
+            super(itemView);
+            mTvFriendName = (TextView) itemView.findViewById(R.id.tvUserName);
+            mBtnAdd = (Button) itemView.findViewById(R.id.btnAdd);
+            mBtnAdd.setOnClickListener(this);
+        }
+
+        @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/ListFriendAdapter.java]: remove this line `@RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN`

File: app/src/main/java/vn/asiantech/internship/ListFriendAdapter.java [hunk 32]
```diff
@@ -0,0 +1,105 @@
...
+        private void setColor() {
+            mBtnAdd.setBackground(mContext.getResources().getDrawable(R.drawable.bg_btn_added_friend));
+            mBtnAdd.setText(mContext.getResources().getString(R.string.Button_Text_Friend));
+            mBtnAdd.setTextColor(mContext.getResources().getColor(R.color.friendsColorWhite));
+            mFriends.get(getLayoutPosition()).setFriend(true);
+            mIsFriend = true;
+        }
+
+        @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/ListFriendAdapter.java]: remove this line `@RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN`

File: app/src/main/res/layout/activity_main.xml [hunk 33]
```diff
@@ -1,19 +1,41 @@
...
-    <TextView
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:text="Hello World!"
-        app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toTopOf="parent"/>
+    <LinearLayout
+        android:id="@+id/recyclerView"
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/activity_main.xml]: Thís is `LinearLayout` why id is `recyclerView`?

**Pull request merged**